kpt-backstage-plugins/plugins/cad/src/components/ResourceEditorDialog/components/FirstClassEditors/KptfileEditor/KptfileEditor.tsx

```diff
@@ -0,0 +1,262 @@
+    () => allKptFunctions.filter(isValidatorFunction),
+    [allKptFunctions],
+  );
+
+  const [expanded, setExpanded] = useState<string>();
+
+  const handleChange =
+    (panel: string) => (_: ChangeEvent<{}>, newExpanded: boolean) => {
+      setExpanded(newExpanded ? panel : undefined);
+    };
+
+  const classes = useEditorStyles();
+
+  useAsync(async (): Promise<void> => {
+    const allFunctions = await api.listCatalogFunctions();
+    setAllKptFunctions(allFunctions);
+  }, []);
+
+  useEffect(() => {
+    const mapToKptFunction = (fn: KptfileFunctionView): KptfileFunction => {
+      const thisFn = omit(fn, 'key');
+      return thisFn;
+    };
+
+    if (!resourceYaml.pipeline) resourceYaml.pipeline = {};
+
+    resourceYaml.metadata.name = state.name;
+    resourceYaml.metadata.namespace = state.namespace || undefined;
+    resourceYaml.metadata.labels = state.labels;
+    resourceYaml.metadata.annotations = state.annotations;
+    resourceYaml.info.description = state.description;
+    resourceYaml.pipeline.mutators =
+      mutators.length > 0 ? mutators.map(mapToKptFunction) : undefined;
+    resourceYaml.pipeline.validators =
+      validators.length > 0 ? validators.map(mapToKptFunction) : undefined;
+
+    if (state.annotations && Object.keys(state.annotations).length === 0) {
+      delete resourceYaml.metadata.annotations;
+    }
+    if (state.labels && Object.keys(state.labels).length === 0) {
+      delete resourceYaml.metadata.labels;
+    }
+
+    onUpdatedYaml(dump(resourceYaml));
+  }, [state, mutators, validators, resourceYaml, onUpdatedYaml]);
+
+  const kptFunctionUpdateFn = (
+    lst: KptfileFunctionView[],
+    setFunctions: React.Dispatch<React.SetStateAction<KptfileFunctionView[]>>,
+    currentFunction: KptfileFunctionView,
+    updatedFunction?: KptfileFunctionView,
+  ) => {
+    const idx = lst.indexOf(currentFunction);
+    const updatedList = lst.slice();
+    if (updatedFunction) {
+      updatedList[idx] = updatedFunction;
+    } else {
+      updatedList.splice(idx, 1);
+    }
+    setFunctions(updatedList);
+  };
+
+  const updateMutatorFunction = (
+    currentFunction: KptfileFunctionView,
+    updatedFunction?: KptfileFunctionView,
+  ): void => {
+    kptFunctionUpdateFn(
+      mutators,
+      setMutators,
+      currentFunction,
+      updatedFunction,
+    );
+  };
+
+  const updateValidatorFunction = (
+    currentFunction: KptfileFunctionView,
+    updatedFunction?: KptfileFunctionView,
+  ): void => {
+    kptFunctionUpdateFn(
+      validators,
+      setValidators,
+      currentFunction,
+      updatedFunction,
+    );
+  };
+
+  return (
+    <div className={classes.root}>
+      <SingleTextFieldAccordion
+        title="Name"
+        expanded={expanded === 'name'}
+        onChange={handleChange('name')}
+        value={state.name}
+        onValueUpdated={value => setState(s => ({ ...s, name: value }))}
+      />
+      <SingleTextFieldAccordion
+        title="Namespace"
```

Comment:
I doubt why we want show the namespace in editor for Kptfile.
Kptfile doesn't require a namespace and it actually doesn't use it anywhere.

Reply:
Removed.